src/transformers/audio_utils.py

```diff
@@ -81,279 +76,521 @@ def mel_to_hertz(mels: np.array, mel_scale: str = "htk") -> np.array:
-          a
-        linear scale. Generally to double the percieved volume of a sound we need to put 8 times as much energy into
-        it.
-        - This means that large variations in energy may not sound all that different if the sound is loud to begin
-        with. This compression operation makes the mel features match more closely what humans actually hear.
+    To create a log-mel spectrogram, call `np.log10(...)` on the output of this function. To convert to true decibels,
+    use `amplitude_to_db()` or `power_to_db()`, depending on whether this is an amplitude or power spectrogram.
 
     Args:
-        mel_spectrogram (`np.array`):
-            Input mel spectrogram.
-        top_db (`int`, *optional*):
-            The maximum decibel value.
-        a_min (`int`, *optional*, default to 1e-10):
-            Minimum value to use when cliping the mel spectrogram.
-        ref (`float`, *optional*, default to 1.0):
-            Maximum reference value used to scale the mel_spectrogram.
+        spectrogram (`np.ndarray` of shape `(num_freq_bins, length)`):
+            The input amplitude or power spectrogram.
+        mel_filters (`np.ndarray` of shape `(num_freq_bins, num_mel_filters)`):
+            The mel filter bank.
+        mel_floor (`float`, *optional*, defaults to 1e-10):
+            Minimum value of mel frequency banks.
+
+    Returns:
+        `np.ndarray` of shape `(num_mel_filters, length)`: The mel spectrogram.
+    """
+    mel = np.maximum(mel_floor, np.dot(mel_filters.T, spectrogram))
+    return mel
 
+
+def optimal_fft_length(window_length: int) -> int:
+    """
+    Finds the best FFT input size for a given `window_length`. This function takes a given window length and, if not
+    already a power of two, rounds it up to the next power or two.
+
+    The FFT algorithm works fastest when the length of the input is a power of two, which may be larger than the size
+    of the window or analysis frame. For example, if the window is 400 samples, using an FFT input size of 512 samples
+    is more optimal than an FFT size of 400 samples. Using a larger FFT size does not affect the detected frequencies,
+    it simply gives a higher frequency resolution (i.e. the frequency bins are smaller).
     """
-    log_spec = 10 * np.log10(np.clip(mel_spectrogram, a_min=a_min, a_max=None))
-    log_spec -= 10.0 * np.log10(np.maximum(a_min, ref))
-    if top_db is not None:
-        if top_db < 0:
-            raise ValueError("top_db must be non-negative")
-        log_spec = np.clip(log_spec, min=np.maximum(log_spec) - top_db, max=np.inf)
-    return log_spec
+    return 2 ** int(np.ceil(np.log2(window_length)))
 
 
-# TODO @ArthurZucker: This method does not support batching yet as we are mainly focus on inference.
-def fram_wave(waveform: np.array, hop_length: int = 160, fft_window_size: int = 400, center: bool = True):
+def window_function(
```

Comment:
Same comment as before on the renaming.

Reply:
@hollance As @sgugger mentioned, we can't *just*  remove/change code without a deprecation cycle when it will break things. Even if it's bad code, it's exposed and users use them. We can't just think about the clean code without considering what our users will face.